cmd/eigentrust/cmd/basiccompute.go

```diff
@@ -140,6 +140,106 @@ func loadInlineLocalTrustCsv(filename string, ref *basic.LocalTrustRef) error {
 	return nil
 }
 
+func trustVectorURIToRef(uri string, ref *basic.TrustVectorRef) error {
+	parsed, err := url.Parse(uri)
+	if err != nil {
+		return err
+	}
+	switch parsed.Scheme {
+	case "file", "":
+		path := parsed.Path
+		if path == "" {
+			path = parsed.Opaque
+		}
+		return loadInlineTrustVector(path, ref)
+	default:
+		return errors.Errorf("invalid trust vector URI scheme %#v",
+			parsed.Scheme)
+	}
+}
+
+func loadInlineTrustVector(filename string, ref *basic.TrustVectorRef) error {
+	logger.Trace().Str("filename", filename).Msg("loading inline trust vector")
+	ext := strings.ToLower(filepath.Ext(filename))
+	switch ext {
+	case ".csv":
+		return loadInlineTrustVectorCsv(filename, ref)
+	default:
+		return errors.Errorf("invalid trust vector file type %#v", ext)
+	}
+}
+
+func loadInlineTrustVectorCsv(
+	filename string, ref *basic.TrustVectorRef,
+) error {
+	f, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer func() { _ = f.Close() }()
+	reader := csv.NewReader(f)
+	inputErrorf := func(field int, format string, v ...interface{}) error {
+		line, column := reader.FieldPos(0)
+		return errors.Errorf("%s:%d:%d: %s", filename, line, column,
+			fmt.Sprintf(format, v...))
+	}
+	inputWrapf := func(
+		err error, field int, format string, v ...interface{},
+	) error {
+		line, column := reader.FieldPos(0)
+		return errors.Wrapf(err, "%s:%d:%d: %s", filename, line, column,
+			fmt.Sprintf(format, v...))
+	}
+	inline := basic.InlineTrustVector{
+		Entries: nil,
+		Scheme:  "inline",
+		Size:    0,
+	}
+	fields, err := reader.Read()
+	for ; err == nil; fields, err = reader.Read() {
+		if len(fields) < 1 {
+			return inputErrorf(0, "too few (%d) fields", len(fields))
+		}
+		if len(fields) > 2 {
+			return inputErrorf(0, "too many (%d) fields", len(fields))
+		}
+		var (
+			from  int64
+			value float64
+		)
+		from, err = strconv.ParseInt(fields[0], 0, 0)
+		switch {
+		case err != nil:
+			return inputWrapf(err, 0, "invalid from=%#v", fields[0])
+		case from < 0:
+			return inputErrorf(0, "negative from=%#v", from)
+		}
+		i := int(from)
+		value, err = strconv.ParseFloat(fields[1], 64)
+		switch {
+		case err != nil:
+			return inputWrapf(err, 1, "invalid trust value=%#v", fields[2])
```

Comment:
This seems like typo that is referencing fields[2] instead of fields[1]

Reply:
Fixed, thanks!